ansible/roles/providers/tasks/main.yml

```diff
@@ -79,14 +78,34 @@
     mode: 0644
   when: ansible_os_family != "Flatcar"
 
+- name: get cloud_init version
+  register: cloud_init_version
+  shell: cloud-init --version | awk '{print $2}' | cut -d- -f1
+  when: ansible_os_family !=  "Flatcar"
+
 - name: set cloudinit feature flags
   copy:
     src: usr/lib/python3/dist-packages/cloudinit/feature_overrides.py
     dest: "{{ sysusr_prefix }}/lib/python3/dist-packages/cloudinit/feature_overrides.py"
     owner: root
     group: root
     mode: 0644
-  when: ansible_os_family == "Debian"
+  when: ansible_os_family == "Debian" and  cloud_init_version.stdout  | float > 20.0
```

Comment:
This here is a special case. Cloud init comes gets installed in dist-packages instead of site-packages.

https://wiki.debian.org/Python#Deviations_from_upstream